pkg/util/push/push.go

```diff
@@ -121,15 +137,59 @@ func handler(maxRecvMsgSize int,
 				level.Warn(logger).Log("msg", "push request canceled", "err", err)
 				return
 			}
+			var (
+				code int
+				msg  string
+			)
 			resp, ok := httpgrpc.HTTPResponseFromError(err)
-			if !ok {
-				http.Error(w, err.Error(), http.StatusInternalServerError)
-				return
+			if ok {
```

Comment:
Can we make this, please, to make sure that `resp` only lives in the conditional block?
```suggestion
			if resp, ok := httpgrpc.HTTPResponseFromError(err); ok {
```